src/symmetric_heap_c.c

```diff
@@ -329,6 +331,8 @@ shmem_realloc(void *ptr, size_t size)
     void *ret;
 
     SHMEM_ERR_CHECK_INITIALIZED();
+
+    if (size == 0) return ptr;
```

Comment:
We may have to do a little more here - "If size is 0 and ptr is not a null pointer, the block to which it points is freed."

Reply:
Thanks @davidozog. That case is already handled further down in the code. However, I changed the if statement here to check if the ptr is also NULL to make sure that the other options get checked. Let me know if you see any issues.

Reply:
Whoops!  Missed that - thanks!